### PR TITLE
Add oracle-net-descriptor option

### PR DESCRIPTION
**Pull request by Michael-A-McMahon**
Adds an extended option named "oracle-net-descriptor" that can specify a descriptor or tnsnames alias.

### CODE REVIEW

File: src/main/java/oracle/r2dbc/impl/OracleConnectionFactoryImpl.java [hunk 1]
```diff
@@ -59,7 +59,19 @@
  *   <li>{@link ConnectionFactoryOptions#DATABASE}</li>
  *   <li>{@link ConnectionFactoryOptions#USER}</li>
  *   <li>{@link ConnectionFactoryOptions#PASSWORD}</li>
- * </ul>
+ *   <li>{@link ConnectionFactoryOptions#CONNECT_TIMEOUT}</li>
+ *   <li>{@link ConnectionFactoryOptions#SSL}</li>
+ * </ul><p>
+ * A value set for {@code CONNECT_TIMEOUT} will be rounded up to the nearest
```
Comment (Douglas-Surber) [inline src/main/java/oracle/r2dbc/impl/OracleConnectionFactoryImpl.java:65]: Would it be easier to read to include these comments in the above list, perhaps using a definition list?

File: src/main/java/oracle/r2dbc/impl/OracleReactiveJdbcAdapter.java [hunk 2]
```diff
@@ -251,53 +270,39 @@ static OracleReactiveJdbcAdapter getInstance() {
...
-   * Database; It is not interpreted as a system identifier (SID).
-   * </p><p>
-   * Values set for {@code USER} and {@code PASSWORD} options are used to
-   * authenticate with an Oracle Database.
+   * When the "descriptor" option is provided, it is invalid to specify any
+   * other options that might conflict with values also specified in the
+   * descriptor. For instance, the descriptor element of
+   * {@code (ADDRESSS=(HOST=...)(PORT=...)(PROTOCOL=...))} specifies values
+   * that overlap with the standard {@code Option}s of {@code HOST}, {@code
+   * PORT}, and {@code SSL}. An {@code IllegalStateException} is thrown
```
Comment (Douglas-Surber) [inline src/main/java/oracle/r2dbc/impl/OracleReactiveJdbcAdapter.java:296]: Would IllegalArgumentException make more sense? Just thinking out loud.

Comment (Michael-A-McMahon) [reply src/main/java/oracle/r2dbc/impl/OracleReactiveJdbcAdapter.java:296]: Yes, I think you are right. Reviewing the spec, this case seems to meet the criteria for IllegalArugmentException:

> Drivers throw IllegalArgumentException if a method has been received an illegal or inappropriate argument (such as values that are out of bounds or an expected parameter is null). This exception is a generic exception that is not associated with an error code or an SQLState

https://r2dbc.io/spec/0.8.4.RELEASE/spec/html/#exceptions

File: src/main/java/oracle/r2dbc/impl/OracleConnectionFactoryImpl.java (hunk 1 above)
Comment (Michael-A-McMahon) [reply src/main/java/oracle/r2dbc/impl/OracleConnectionFactoryImpl.java:65]: Thanks to this comment, I've now learned something new about HTML:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl
 
I will use <dl> here, and probably start using it in many other places too.

Comment (Michael-A-McMahon) [reply src/main/java/oracle/r2dbc/impl/OracleConnectionFactoryImpl.java:65]: Also, any HTML we type in a comment will be rendered. Now I can try out this new tag:
<dl>
<dt>CONNECT_TIMEOUT</dt>
<dd>The duration will be rounded up to the nearest whole second</dd>
</dl>

**Comment by Michael-A-McMahon** [comment]
Latest commit changes the option name from "oracle-net-descriptor" to "oracleNetDescriptor". The camelCase form seems to be more conventional.

**Review by Douglas-Surber — APPROVED** [review]

File: src/main/java/oracle/r2dbc/impl/OracleReactiveJdbcAdapter.java (hunk 2 above)
Comment (netlineup) [reply src/main/java/oracle/r2dbc/impl/OracleReactiveJdbcAdapter.java:296]: You still have IllegalStateException mentioned here.

Comment (jeandelavarene) [reply src/main/java/oracle/r2dbc/impl/OracleReactiveJdbcAdapter.java:296]: @Michael-A-McMahon you still have IllegalStateException mentioned in the javadoc (line 296). This needs to be changed to IllegalArugmentException.

Comment (Michael-A-McMahon) [reply src/main/java/oracle/r2dbc/impl/OracleReactiveJdbcAdapter.java]: Thanks for catching that, it's now fixed in the latest commit

**Pull request merged**